clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -258,7 +258,10 @@ public void poll(long now) {
         }
 
         pollHeartbeat(now);
-        maybeAutoCommitOffsetsAsync(now);
+        if (subscriptions.hasNoSubscription())
+            subscriptions.clearAssignment();
```

Comment:
Thanks. Yes it makes sense, I just wanted to be clear about the `unsubscribe()` partition assignment semantics. I'll update the patch so that assigned partitions are also cleared immediately when calling `unsubscribe()`.